src/app/event/Modal.tsx

```diff
@@ -0,0 +1,117 @@
+import { useState } from "react";
+import { isAddress } from "viem";
+import {
+  Dialog,
+  DialogContent,
+  DialogDescription,
+  DialogHeader,
+  DialogTitle,
+  DialogTrigger,
+} from "@/components/ui/dialog";
+import { useFieldArray, useForm } from "react-hook-form";
+
+import {
+  AssignTicketText,
+  CloseDialogButton,
+  TriggerDialogButton,
+} from "./components/ModalComponents";
+
+interface FormData {
+  addresses: { address: string }[];
+}
+
+export function Modal() {
+  const [open, setOpen] = useState(false);
+  const {
+    register,
+    control,
+    handleSubmit,
+    watch,
+    formState: { errors, isValid },
+  } = useForm<FormData>({
+    defaultValues: {
+      addresses: [{ address: "" }],
+    },
+    mode: "onChange",
+  });
+
+  const { fields, append, remove } = useFieldArray({
+    control,
+    name: "addresses",
+  });
+
+  const watchedFields = watch("addresses");
+
+  function submitForm(data: FormData) {
+    const filteredData = data.addresses.filter(
+      (element) => element.address !== ""
+    );
+    console.log("Form submitted with data:", filteredData);
+    setOpen(false);
+  }
+
+  function handleNumberOfFields() {
+    const lastElement = watchedFields[watchedFields.length - 1];
+    if (isAddress(lastElement?.address)) {
+      append({ address: "" });
+    }
+
+    const emptyElements = watchedFields
+      .map((field, index) => (field.address === "" ? index : -1))
+      .filter((index) => index !== -1);
+    if (emptyElements.length > 1) {
+      remove(emptyElements[0]);
+    }
+  }
```

Comment:
I would add an useCallback here, probably it doens't add a lot of performance in this specific case, but IMO this type of function where onChange is called multiple times, should have a memorized version.

```suggestion
  const handleNumberOfFields = useCallback(() => {
    const lastElement = watchedFields[watchedFields.length - 1];
    if (isAddress(lastElement?.address)) {
      append({ address: "" });
    }
    const emptyElements = watchedFields
      .map((field, index) => (field.address === "" ? index : -1))
      .filter((index) => index !== -1);
    if (emptyElements.length > 1) {
      remove(emptyElements[0]);
    }
  }, [watchedFields, append, remove]);
```